packages/static-build/src/utils/gatsby.ts

```diff
@@ -335,3 +335,27 @@ module.exports = gatsbyNode;
 `
   );
 }
+
+export async function cleanupGatsbyFiles(dir: string) {
+  const backup = '.__vercel_builder_backup__';
+  const fileEndings = ['.js', '.ts', '.mjs'];
+
+  for (const fileName of [GATSBY_CONFIG_FILE, GATSBY_NODE_FILE]) {
+    for (const fileEnding of fileEndings) {
+      const backupFile = `${fileName}${fileEnding}${backup}${fileEnding}`;
+      const backupFilePath = path.join(dir, backupFile);
+      const generatedFile = `${fileName}${fileEnding}`;
+      const generatedFilePath = path.join(dir, generatedFile);
+
+      if (
+        (await fileExists(backupFilePath)) &&
+        (await fileExists(generatedFilePath))
+      ) {
+        await fs.rm(generatedFilePath);
+        await fs.rename(backupFilePath, generatedFilePath);
+      } else if (await fileExists(generatedFilePath)) {
+        await fs.rm(generatedFilePath);
```

Comment:
This one seems dangerous. How about if the user existing config file was not modified?

I think we may want to add a comment to the top of the generated files and check for it before deleting.